language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/MethodCallExpressionNodeContext.java

```diff
@@ -57,10 +58,13 @@ public boolean onPreValidation(BallerinaCompletionContext context, MethodCallExp
         hierarchy and skip properly
         eg:
         (2) s<cursor>abc.def.testMethod()
+        (3) self.<cursor>Method()
          */
         int cursor = context.getCursorPositionInTree();
         NameReferenceNode nameRef = node.methodName();
+        Token dotToken = node.dotToken();
 
-        return cursor >= nameRef.textRange().startOffset() && cursor <= nameRef.textRange().endOffset();
+        return ((cursor >= nameRef.textRange().startOffset() && cursor <= nameRef.textRange().endOffset())
+                || (!dotToken.isMissing() && cursor > dotToken.textRange().startOffset()));
```

Comment:
What's the sample we are going to cover with this additional check?

Reply:
@nadeeshaan The sample would be `start self.<cursor>` . It failed on `onPreValidation()` in MethodCallExpression, and then picked up the provider incorrectly as `StartActionNodeContext` . Added this check to avoid that.

Reply:
I was wondering whether this should direct to `FieldAccessExpressionNodeContext`. Can you add a comment in this place?

Reply:
@nadeeshaan Yes, that did occur to me. But since we should only allow methods in the `start` action expression, I thought it would be alright directing to `MethodCallExpressionNodeContext` . I have added a comment in line [#61](https://github.com/ballerina-platform/ballerina-lang/pull/31210/files#diff-998e23560dea026640124f3c72c1a969d50f5e3cb847da96f3b72bc78a10e231R61) regarding this. Would it work?